src/widget/core.rs

```diff
@@ -172,7 +177,12 @@ impl Pod {
     pub fn mark(&mut self, flags: ChangeFlags) -> ChangeFlags {
         self.state
             .request(PodFlags::from_bits_truncate(flags.bits as _));
-        ChangeFlags::from_bits_truncate(self.state.upwards_flags().bits as _)
+        let mut flags_up =
+            flags & ChangeFlags::from_bits_truncate(PodFlags::UPWARD_FLAGS.bits as _);
+        if flags.contains(ChangeFlags::ACCESSIBILITY) {
+            flags_up |= ChangeFlags::DESCENDANT_REQUESTED_ACCESSIBILITY;
+        }
+        flags_up
```

Comment:
Same as `merge_up`

Reply:
The first one I agree with, but this one doesn't have quite the same logic. The flags propagated up the `rebuild` traversal aren't the ones from the `Pod` - I think that was actually a bug in the previous implementation that this PR fixes. And `ChangeFlags` is a different type than `PodFlags`. That said, I could look into making methods to improve DRY.